kubernetes/schema_pod_spec.go

```diff
@@ -385,6 +385,42 @@ func podSpecFields(isUpdatable, isComputed bool) map[string]*schema.Schema {
 				},
 			},
 		},
+		"topology_spread_constraint": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "describes how a group of pods ought to spread across topology domains. Scheduler will schedule pods in a way which abides by the constraints.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"max_skew": {
+						Type:         schema.TypeInt,
+						Description:  "describes the degree to which pods may be unevenly distributed.",
+						Optional:     true,
+						Default:      1,
+						ValidateFunc: validation.IntNotInSlice([]int{0}),
```

Comment:
I think here you want [validation.IntAtLeast(1)](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/validation#IntAtLeast) as the the int can't be a negative number.

The docs for this field are not super clear, but if you look at the code the `<= 0` condition is here: https://github.com/kubernetes/kubernetes/blob/e059af640627c7be36b00c1d10a4d511e283d9b4/pkg/scheduler/apis/config/validation/validation_pluginargs.go#L134